app/components/dynamic-select/dynamic-select.tsx

```diff
@@ -148,7 +148,9 @@ export default function DynamicSelect({
 
         <Popover open={isPopoverOpen} onOpenChange={setIsPopoverOpen}>
           <PopoverTrigger disabled={disabled} asChild>
-            <div>
+            <button
+              className={disabled ? "cursor-not-allowed opacity-60" : "w-full"}
```

Comment:
We need to check the other places where we use DynamicSelect and make sure it didnt break the UI

Reply:
I have already checked it and it is working fine.